### ISSUE TITLE
Add support to specify database for mongoose.model()

### ISSUE THREAD

**Issue by useo6** [Automattic/mongoose#6627]
- [x]  feature
- [ ] bug

**What is the current behavior?**

I created a MongoDB cluster using MongoDB Atlas. I created two databases inside my cluster because I want separate my backend and frontend. Now, I've a user to access the frontend and another user to access the backend. The frontend user can access the frontend database only. The backend user can access both databases.
Both users are using the same (except login credentials) connection string because both databases are in the same cluster. The problem is that I cannot specify the database when creating my models using mongoose.model(). Hence, at the moment I've to do something like this:

```
let backendConnection = mongoose.createConnection('mongodb//localhost/backenddb')
let frontendConnection = mongoose.createConnection('mongodb//localhost/frongenddb')

let ProductModel= frontendConnection.model('Product', new mongoose.Schema({name: String}))
let Order = backendConnection.model('Order', new mongoose.Schema({product: {mongoose.Schema.Types.ObjectId, ref: 'Product'}}))

Order.find().populate({path: 'product', model: ProductModel}, console.log)
```

So, I've multiple connections to the same server (but different databases), need to store the connections somewhere (at the moment I'm doing it in the globals) and need to specifiy the model when populating a model from another connection (while I think I can still use the same connection because in case of the backend user I can access the frontend AND backend database).

**What is the expected behavior?**

What I imagine is something like this:

```
mongoose.connect('mongodb://localhost')

let Product= mongoose.model('Product', new mongoose.Schema({name: String}), '<NameOfFrontendDatabase>')
let Order= mongoose.model('Order', new mongoose.Schema({product: {type: mongoose.Schema.Types.ObjectId, ref: 'Product'}}), '<NameOfBackendDatabase>')

Order.find().populate('product', console.log)

```

No need to use multiple connections, no need to specify the model when populating, no need to keep the connections in the globals or somewhere else (the names of the databases can be stored in environment varaibles).

mongoose Version: 5.0.15
NodeJS Version: 8.10.0

**Comment by vkarpov15** [comment]
Reasonable suggestion, will keep this open for the future, but this isn't likely to happen soon because right now mongoose connections are tied closely to a single database, and supporting multiple dbs off of one connection will be tricky.

**Comment by vkarpov15** [comment]
AFAIK this should be fixed by #6520, which was released with 5.6.0 :+1:
